### PR TITLE
Badoken/global enum naming strategy

### LINKED ISSUE #4674: Allow setting global enum naming strategy similar to property naming strategy
### Is your feature request related to a problem? Please describe.

- Currently you have to annotate all enums with `EnumNaming` in order to change default enum naming strategy which is cumbersome and verbose.

### Describe the solution you'd like

It should be possible to set default enum naming strategy directly on `ObjectMapper`.

### Usage example

`ojbectMapper.setEnumNamingStrategy(EnumNamingStrategies.CAMEL_CASE)`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by badoken**
Fixes/implements #4674 
Replaces #4716

### CODE REVIEW

File: src/main/java/com/fasterxml/jackson/databind/cfg/BaseSettings.java [hunk 1]
```diff
@@ -67,6 +67,11 @@ public final class BaseSettings
      */
     protected final PropertyNamingStrategy _propertyNamingStrategy;
 
+    /**
+     * Custom enum naming strategy in use, if any.
```
Comment (JooHyukKim) [inline src/main/java/com/fasterxml/jackson/databind/cfg/BaseSettings.java:71]:
```suggestion
     * Custom enum naming strategy in use, if any.
     *
     * @since 2.19 
```

File: src/main/java/com/fasterxml/jackson/databind/cfg/BaseSettings.java [hunk 2]
```diff
@@ -157,10 +162,10 @@ public final class BaseSettings
      */
 
     /**
-     * @since 2.16
+     * @since 2.19
      */
     public BaseSettings(ClassIntrospector ci, AnnotationIntrospector ai,
-            PropertyNamingStrategy pns, TypeFactory tf,
+            PropertyNamingStrategy pns, EnumNamingStrategy ens, TypeFactory tf,
```
Comment (JooHyukKim) [inline src/main/java/com/fasterxml/jackson/databind/cfg/BaseSettings.java:168]: Due to backward-compatibility issue, Should write another constructor not modify the existing.
As you wrote comment below, write a new variant that takes {@link EnumNamingStrategy} instead.

```java
     * @deprecated Since 2.19, use variant that takes {@link EnumNamingStrategy} instead.
```

File: src/main/java/com/fasterxml/jackson/databind/introspect/EnumNamingStrategyFactory.java [hunk 3]
```diff
@@ -33,10 +33,34 @@ private EnumNamingStrategyFactory() {}
...
+     *                                  not a subclass of {@link EnumNamingStrategy}.
+     *
+     * @since 2.19
+     */
+    public static EnumNamingStrategy createEnumNamingStrategyInstance(Object namingDef,
+                                                                      boolean canOverrideAccessModifiers,
+                                                                      EnumNamingStrategy defaultNamingStrategy) {
         if (namingDef == null) {
-            return null;
+            return defaultNamingStrategy;
```
Comment (JooHyukKim) [inline src/main/java/com/fasterxml/jackson/databind/introspect/EnumNamingStrategyFactory.java:63]: This behavior does not match the documentation wrt null handling. You are already passing `null` from the deprecated API `createEnumNamingStrategyInstance`

File: src/main/java/com/fasterxml/jackson/databind/cfg/BaseSettings.java (hunk 1 above)
Comment (badoken) [reply src/main/java/com/fasterxml/jackson/databind/cfg/BaseSettings.java:71]: Thanks, updated

File: src/main/java/com/fasterxml/jackson/databind/introspect/EnumNamingStrategyFactory.java (hunk 3 above)
Comment (badoken) [reply src/main/java/com/fasterxml/jackson/databind/introspect/EnumNamingStrategyFactory.java:63]: Thanks, corrected the doc:
```java
     * @return an instance of {@link EnumNamingStrategy} if {@code namingDef} is a subclass of {@link EnumNamingStrategy},
     * {@code defaultNamingStrategy} if {@code namingDef} is {@code null},
     * and an instance of {@link EnumNamingStrategy} if {@code namingDef} already is one.

```

File: src/main/java/com/fasterxml/jackson/databind/cfg/BaseSettings.java (hunk 2 above)
Comment (badoken) [reply src/main/java/com/fasterxml/jackson/databind/cfg/BaseSettings.java:170]: The previous constructor was left unchanged, it's on [line 197](https://github.com/FasterXML/jackson-databind/pull/4723/files#diff-0e1933c5c67aaf7756c31c1a6faeb671597685b4444aa7e4cfa2b8f181359f7cR197).
This constructor is a new one with the additional parameter.

The change is just showed weirdly here

Comment (cowtowncoder) [reply src/main/java/com/fasterxml/jackson/databind/cfg/BaseSettings.java:170]: Verified, true.

**Comment by cowtowncoder** [comment]
I think I am happy with this PR (although slightly worried wrt merging it to main/3.0 but that's not something that can be avoided here). But one last thing before merging: unless we have asked for and received it, we'll need CLA from here:

https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf

This only needs to be done once, and is good for all future PRs.
The usual way is to print it, fill & sign, scan/photo, email to `cla` at fasterxml dot com.
Once this is done, I'll go ahead and merge pr.

Thank you again for this contribution @badoken !

**Comment by badoken** [comment]
Hey @cowtowncoder
Thank you, I'm glad to help
I've just sent the CLA to the email address, please let me know if all is good

**Comment by cowtowncoder** [comment]
CLA received, can proceed!

**Review by cowtowncoder — APPROVED** [review]

File: release-notes/CREDITS-2.x [hunk 4]
```diff
@@ -1839,10 +1839,13 @@ Maxim Valeev (@MaximValeev)
...
  * Contributed fix for #2461: Nested `@JsonUnwrapped` property names not correctly handled
   (2.19.0)
 
+Konstantin Maliuga (@badoken)
+ * Contributed #4674: Allow setting global enum naming strategy similar to property naming strategy
+  (2.19.0)
+
 Lars Benedetto (@lbenedetto)
- * Contributed #4676: Support other enum naming strategies than camelCase
+ * Contributed #4676: Support other enum naming strategies than camelCas
```
Comment (rymiel) [inline release-notes/CREDITS-2.x:1850]: Is this change intentional?

Comment (cowtowncoder) [reply release-notes/CREDITS-2.x:1850]: Nope. Thanks for pointing out!
